src/dhtproto/client/request/internal/GetHashRange.d

```diff
@@ -208,229 +193,197 @@ public struct GetHashRange
-
-        Next state:
-            - Initialising (once connection is established)
+        FillPayload policy, called from AllNodesRequestInitialiser template
+        to add request-specific data to the initial message payload send to the
+        node to begin the request.
 
-        Returns:
-            next state
+        Params:
+            payload = message payload to be filled
 
     ***************************************************************************/
 
-    private State stateEstablishingConnection ( )
+    private void fillPayload ( RequestOnConnBase.EventDispatcher.Payload payload )
     {
-        while (true)
-        {
-            switch (this.conn.waitForReconnect())
-            {
-                case conn.FiberResumeCodeReconnected:
-                case 0: // The connection is already up
-                    return State.Initialising;
-
-                default:
-                    assert(false,
-                        typeof(this).stringof ~ ".stateWaitingForReconnect: " ~
-                        "Unexpected fiber resume code when reconnecting");
-            }
-        }
+        // Nothing more to add. (Request code and version already added.)
     }
 
     /***************************************************************************
 
-        Sends the request code, version, etc. to the node to begin the request,
-        then receives the status code and hash range from the node.
+        HandleStatusCode policy, called from AllNodesRequestInitialiser
+        template to decide how to handle the status code received from the node.
 
-        Next state:
-            - Receiving (once the request is initialised)
-            - Exit, if the node returns an error status
+        Params:
+            supported = supported code received from the node in response to the
+                initial message
 
         Returns:
-            next state
+            true to continue handling the request (supported); false to abort
+            (unsupported)
 
     ***************************************************************************/
 
-    private State stateInitialising ( )
-    in
+    private bool handleSupportedCode ( ubyte code )
     {
-        // stateWaitingForReconnect should guarantee we're already connected
-        assert(this.conn.waitForReconnect() == 0);
-    }
-    body
-    {
-        // Send request info to node
-        this.conn.send(
-            ( conn.Payload payload )
-            {
-                payload.add(GetHashRange.cmd.code);
-                payload.add(GetHashRange.cmd.ver);
-            }
-        );
-        this.conn.flush();
-
-        // Receive status from node and stop the request if not Ok
-        auto status = conn.receiveValue!(StatusCode)();
-        switch ( status )
+        auto supported = cast(SupportedStatus)code;
+        if ( !GetHashRange.handleSupportedCodes(supported,
+            this.context, this.conn.remote_address) )
```

Comment:
Perhaps in the next swarm we can adapt the `AllNodesRequest` to pass the `SupportedStatus` to this policy method.

Reply:
(or adapt it to avoid the policy all together).

Reply:
Yeah, we can probably provide something standard.